obfuscation_detection/heuristics.py

```diff
@@ -122,3 +122,41 @@ def find_xor_decryption_loops(bv):
         if contains_xor_decryption_loop(bv, f):
             print(
                 f"Function {hex(f.start)} ({f.name}) contains a XOR decryption loop with a constant.")
+
+arithmetic_op = [HighLevelILOperation.HLIL_ADD, HighLevelILOperation.HLIL_NEG, HighLevelILOperation.HLIL_SUB, HighLevelILOperation.HLIL_MUL, HighLevelILOperation.HLIL_DIVS, HighLevelILOperation.HLIL_MODS]
+boolean_op = [HighLevelILOperation.HLIL_NOT, HighLevelILOperation.HLIL_AND, HighLevelILOperation.HLIL_OR, HighLevelILOperation.HLIL_XOR]
+arithmetic_counter = 0
+boolean_counter = 0
+
+def traverse_HLIL(il, ident):
+    global arithmetic_counter, boolean_counter
+    if isinstance(il, highlevelil.HighLevelILInstruction):
+        if il.operation in arithmetic_op:
+            arithmetic_counter += 1
+
+        if il.operation in boolean_op:
+            boolean_counter += 1
+
+        for o in il.operands:
+            traverse_HLIL(o, ident+1)
+
+def find_mba_expressions(bv):
+    print("=" * 80)
+    print("Functions that might have MBA expressions:")
+    global boolean_counter, arithmetic_counter
+    # iterate through functions 
+    for f in bv.functions:
+        instr_mba = 0
+        if f.hlil is not None:
+            # iterate through HIL instructions via AST
+            for ins in f.hlil.root:
+                # reset the counters per each instruction
```

Comment:
> use a set instead of lists for loopup; this makes lookups in the worst case O(1) instead of O(n)

I hope I got it right! Otherwise feel free to comment

Fixed 2. and 3.